eva-stats/stats_accessioning.py

```diff
@@ -0,0 +1,100 @@
+
+
+logger = logging_config.get_logger(__name__)
+
+
+def get_handles(private_config_xml_file):
+    properties = get_properties_from_xml_file("production", private_config_xml_file)
+    mongo_hosts_and_ports = str(properties['eva.mongo.host'])
+    mongo_host, mongo_port = get_mongo_primary_host_and_port(mongo_hosts_and_ports)
+    mongo_username = str(properties['eva.mongo.user'])
+    mongo_password = str(properties['eva.mongo.passwd'])
+
+    mongo_handle = get_mongo_connection_handle(mongo_username, mongo_password, mongo_host)
+    metadata_handle = psycopg2.connect(get_pg_metadata_uri_for_eva_profile(
+        "development", private_config_xml_file), user="evadev")
+
+    return mongo_handle, metadata_handle
+
+
+def get_mongo_primary_host_and_port(mongo_hosts_and_ports):
+    """
+    :param mongo_hosts_and_ports: All host and ports stored in the private settings xml
+    :return: mongo primary host and port
+    """
+    for host_and_port in mongo_hosts_and_ports.split(','):
+        if '001' in host_and_port:
+            properties = host_and_port.split(':')
+            return properties[0], properties[1]
+
+
+def get_stats_from_accessioning_db(mongo_handle, metadata_handle, provided_assemblies):
+    logger.info("Getting counts for RS IDs (Clustered Variants)")
+    store_accessioning_counts(mongo_handle, metadata_handle, 'clusteredVariantEntity', provided_assemblies)
+    logger.info("Getting counts for SS IDs (Submitted Variants)")
+    store_accessioning_counts(mongo_handle, metadata_handle, 'submittedVariantEntity', provided_assemblies)
+
+
+def store_accessioning_counts(mongo_handle, metadata_handle, collection_name, provided_assemblies):
+    collection = mongo_handle["eva_accession_sharded"][collection_name]
+    assembly_field = "asm" if collection_name == "clusteredVariantEntity" else "seq"
+    stats_table = "rs_stats" if collection_name == "clusteredVariantEntity" else "ss_stats"
+    assemblies = get_assemblies(provided_assemblies, assembly_field, collection)
+    for assembly in assemblies:
+        pipeline = [
+            {"$match": {assembly_field: assembly}},
+            {"$group": {"_id": assembly, "count": {"$sum": 1}}}
+        ]
+        cursor_rs = collection.aggregate(pipeline)
+        for rs_stat in cursor_rs:
+            logger.info(rs_stat)
+            assembly = rs_stat["_id"]
+            count = rs_stat["count"]
+            query = "insert into eva_stats.{2} values ('{0}', {1}) " \
+                    "on conflict(assembly) do update set num_variants = {1}".format(assembly, count, stats_table)
+            execute_query(metadata_handle, query)
+
+
+def get_assemblies(provided_assemblies, assembly_field, collection):
+    if provided_assemblies:
+        assemblies = []
+        for assembly in provided_assemblies.split(','):
+            assemblies.append(assembly)
+        return assemblies
+    else:
+        return collection.distinct(assembly_field)
```

Comment:
As more assemblies are added, this might time out in the future as well :( but this operation uses an index so it might be a while before that happens!   If it does time out in the future, we should maintain a separate registry of accessioned assemblies somewhere!

Reply:
I agree. The long term solution should probably be to integrate the counts in the submission automation so counts get updated when we are accessioning variants.